src/rust/perftools/profiler/mod.rs

```diff
@@ -302,12 +307,29 @@ impl Profiler {
         let total_duration = self.roots.iter().map(|root| root.borrow().duration_sum).sum();
 
         for root in self.roots.iter() {
-            root.borrow().write_recursive(out, total_duration, 0, max_depth)?;
+            root.borrow().write_recursive(out, total_duration, 0, max_depth, self.ns_per_cycle)?;
         }
 
         out.flush()
     }
 
+    fn measure_ns_per_cycle() -> f64 {
+        let start = SystemTime::now();
+        let (start_cycle, _): (u64, u32) = unsafe { x86::time::rdtscp() };
+        
+        test::black_box((0..10000).fold(0, |old, new| old ^ new)); // dummy calculations for measurement
+
+        let (end_cycle, _): (u64, u32) = unsafe { x86::time::rdtscp() };
+        let since_the_epoch = SystemTime::now()
```

Comment:
Could you please annotate the following variable declaration with the concerned type?